public/includes/upvote.php

```diff
@@ -0,0 +1,57 @@
+<?php 
+require("config.php");
+$conn = new mysqli($servername, $username , $passd , $dbname);
+if($conn->connect_error){
+	die("Connection failed: " . $conn->connect_error);
+}
+$uid = $_REQUEST["uid"];
+$gid = $_REQUEST["gid"];
+if (preg_match("/[^0-9]/", $uid)===0 &&  preg_match("/[^0-9]/", $gid) ===0){
+	$sql = "select upvote  From games where gameid ='$gid'";
+	$result = $conn->query($sql);
+	if ($result->num_rows === 1){
+		$gdata = $result->fetch_assoc();
+		if (substr_count($gdata["upvote"],"&$uid&")===0){
+			$sql1 = "update games set upvote = concat(upvote, '$uid' , '&') where gameid = $gid";
+			if ($conn->query($sql1)===TRUE){ 
+				$result2 = $conn->query($sql);
+				if ($result2->num_rows === 1){
+					$gdata2 = $result2->fetch_assoc();
+					$u = substr_count($gdata2["upvote"],"&");
+					$n = $u -2;
+					if ($u ===2){
+						echo "You have upvoted the game.";
+					}
+					elseif ($u ===3) {
+						echo "You and 1 other upvoted the game.";
+					}
+					else{
+						echo "You and $n others upvoted the game.";
+					};
```

Comment:
Keep the names of the variables meaningful so it is easier to understand for whoever is going through your code base. In this case $u and $n don't make much sense to me at first glance